### PR TITLE
fix kubectl alpha debug node does not work on tainted(NoExecute) nodes

### LINKED ISSUE #98253: kubectl alpha debug node does not work on tainted nodes
**What happened**:

`kubectl alpha debug node` does not work for nodes with NoExecute taints.

```console
$ kubectl alpha debug node/mynode -it --image=ubuntu
Creating debugging pod node-debugger-mynode-mnn2s with container debugger on node mynode.
Error from server (NotFound): pods "" not found
```

We have several types of Node hardware, and some of them are tainted
to keep away normal pods.

**What you expected to happen**:

A debug pod is created with tolerations for the taints.

**How to reproduce it (as minimally and precisely as possible)**:

1. Add a NoExecute taint to a node
2. Run `kubectl alpha debug node` on the node

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.19.7
- OS (e.g: `cat /etc/os-release`):
    ```
    NAME="Flatcar Container Linux by Kinvolk"
    ID=flatcar
    ID_LIKE=coreos
    VERSION=2605.11.0
    VERSION_ID=2605.11.0
    BUILD_ID=2021-01-08-2026
    PRETTY_NAME="Flatcar Container Linux by Kinvolk 2605.11.0 (Oklo)"
    ANSI_COLOR="38;5;75"
    HOME_URL="https://flatcar-linux.org/"
    BUG_REPORT_URL="https://issues.flatcar-linux.org"
    FLATCAR_BOARD="amd64-usr"
    ```
- Kernel (e.g. `uname -a`): `Linux hostname 5.4.87-flatcar #1 SMP Fri Jan 8 19:54:07 -00 2021 x86_64 Intel(R) Xeon(R) Silver 4114 CPU @ 2.20GHz GenuineIntel GNU/Linux`
- Install tools: CKE
- Network plugin and version (if this is a network-related bug): Coil v2.0.5
- Others:

### PR DESCRIPTION
**Pull request by wawa0210**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

If the node is tainted by `NoExecute`, the debug node will fail, and the debug pod creation will fail, because the debug pod does not consider the node tolerance.

But when debugging a node, we should consider the scenario where the node contains taint

We expect the debug node to work all the time and tolerate everything.

**Which issue(s) this PR fixes**:

Fixes #98253

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix kubectl alpha debug node does not work on tainted(NoExecute) nodes and tolerate everything.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go [hunk 1]
```diff
@@ -462,9 +462,23 @@ func (o *DebugOptions) generateNodeDebugPod(node string) *corev1.Pod {
...
+	// If the current node has the stain of `NoExecute`, in order to debug the pod to run normally,
+	// we need to add its tolerance to the debug pod
+	tolerations := []corev1.Toleration{}
+	for _, taint := range node.Spec.Taints {
+		if taint.Effect == corev1.TaintEffectNoExecute {
+			tolerations = append(tolerations, corev1.Toleration{
+				Key:      taint.Key,
+				Value:    taint.Value,
+				Effect:   taint.Effect,
+				Operator: corev1.TolerationOpExists,
```
[the remark this answers is not in the document]
Comment (ymmt2005) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:479]: @verb 
Yes, `operator: exists` can tolerate any taints regardless of their keys, values, and effects.

[the remark this answers is not in the document]
Comment (wawa0210) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:479]: If this is the case, is it right here to tolerate everything without judging whether the current node contains `taints`, is my understanding correct?

File: staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go [hunk 2]
```diff
@@ -462,9 +462,23 @@ func (o *DebugOptions) generateNodeDebugPod(node string) *corev1.Pod {
...
 	if !o.Quiet {
-		fmt.Fprintf(o.Out, "Creating debugging pod %s with container %s on node %s.\n", pn, cn, node)
+		fmt.Fprintf(o.Out, "Creating debugging pod %s with container %s on node %s.\n", pn, cn, node.Name)
+	}
+
+	// If the current node has the stain of `NoExecute`, in order to debug the pod to run normally,
+	// we need to add its tolerance to the debug pod
+	tolerations := []corev1.Toleration{}
+	for _, taint := range node.Spec.Taints {
+		if taint.Effect == corev1.TaintEffectNoExecute {
```
[the remark this answers is not in the document]
Comment (wawa0210) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go]: Thanks for comments, updated

File: staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (wawa0210) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go]: The current processing exactly matches the `taint` on the node. Every time `debug node` will get the latest `taint` of the node, and tolerate accordingly

[the remark this answers is not in the document]
Comment (verb) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go]: Ok, I think we should simply use `operator: exists` with empty key/values to tolerate everything.

[the remark this answers is not in the document]
Comment (wawa0210) [reply staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go]: updated.

**Review by verb — APPROVED** [review]
/lgtm

**Review by soltysh — APPROVED** [review]
/approve
/priority backlog

**Pull request merged**